lib/action_view/component/content_for.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentFor
+      def content_for(attr, content = nil, options = {}, &block)
+        unless respond_to?(attr)
```

Comment:
Wouldn't we want to be checking for the presence of a setter, not just a reader?

This makes me think we should require these "content_areas" to be registered on the component. If we do that, it might make sense to rename the existing `content` accessor to `body`, so we can have something like:

`content_areas :header, :body, :footer`, and then explicitly check against this list when rendering named blocks.

Reply:
My thinking was that attr_reader is enough because we can set the instance variable behind the attribute. I'm ok with registration. That way we can check/require getter and setter mechanics once